src/Umbraco.Core/Runtime/SqlMainDomLock.cs

```diff
@@ -0,0 +1,407 @@
+
+                        _logger.Debug<SqlMainDomLock>("Timeout elapsed, assuming orphan row, acquiring MainDom.");
+
+                        try
+                        {
+                            db.BeginTransaction(IsolationLevel.ReadCommitted);
+
+                            _sqlServerSyntax.WriteLock(db, Constants.Locks.MainDom);
+
+                            // so now we update the row with our appdomain id
+                            InsertLockRecord(_lockId);
+                            _logger.Debug<SqlMainDomLock>("Acquired with ID {LockId}", _lockId);
+                            return true;
+                        }
+                        catch (Exception ex)
+                        {
+                            ResetDatabase();
+
+                            if (IsLockTimeoutException(ex))
+                            {
+                                // something is wrong, we cannot acquire, not much we can do
+                                _logger.Error<SqlMainDomLock>(ex, "Sql timeout occurred, could not forcibly acquire MainDom.");
+                                _hasError = true;
+                                return false;
+                            }
+                            _logger.Error<SqlMainDomLock>(ex, "Unexpected error, could not forcibly acquire MainDom.");
+                            _hasError = true;
+                            return false;
+                        }
+                        finally
+                        {
+                            db?.CompleteTransaction();
+                        }
+                    }
+                }
+            }, _cancellationTokenSource.Token);
+        }
+
+        /// <summary>
+        /// Inserts or updates the key/value row 
+        /// </summary>
+        private RecordPersistenceType InsertLockRecord(string id)
+        {
+            var db = GetDatabase();
+            return db.InsertOrUpdate(new KeyValueDto
+            {
+                Key = MainDomKey,
+                Value = id,
+                Updated = DateTime.Now
+            });
+        }
+
+        /// <summary>
+        /// Checks if the DB row value is equals the value
+        /// </summary>
+        /// <returns></returns>
+        private bool IsMainDomValue(string val)
+        {
+            var db = GetDatabase();
+            return db.ExecuteScalar<int>("SELECT COUNT(*) FROM umbracoKeyValue WHERE [key] = @key AND [value] = @val",
+                new { key = MainDomKey, val = val }) == 1;
+        }
+
+        /// <summary>
+        /// Checks if the exception is an SQL timeout
+        /// </summary>
+        /// <param name="exception"></param>
+        /// <returns></returns>
+        private bool IsLockTimeoutException(Exception exception) => exception is SqlException sqlException && sqlException.Number == 1222;
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
```

Comment:
```suggestion
        private bool _disposedValue = false; // To detect redundant calls
```

Should follow normal naming convensions